src/lib/ipsec/aes_128_gcm_avx.dasl

```diff
@@ -497,9 +497,9 @@ local mcode, size = Dst:build()
 local entry = dasm.globals(globals, globalnames)
 local fn_t = ffi.typeof("void(*)(gcm_data*, uint8_t*, const uint8_t*, uint64_t, uint8_t*, const uint8_t*, uint64_t, uint8_t*, uint64_t)")
 return setmetatable({
-  aes_keyexp_128_enc_avx = ffi.cast("void(*)(void*, void*)", entry.aes_keyexp_128_enc_avx),
+  aes_keyexp_128_enc_avx = ffi.cast("void(*)(uint8_t*, gcm_data*)", entry.aes_keyexp_128_enc_avx),
```

Comment:
This function really takes `gcm_data.expanded_keys` as its second argument (which itself has no type attached to it besides `uint8_t`. My reasoning here is that since the `expanded_keys` field is the first field of the `gcm_data` struct the pointer to a `gcm_data` object will be the same as the pointer to `gcm_data.expanded_keys`.

Is this a reasonable assumption or is this insane?